pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -153,3 +189,103 @@ func createVeleroBackupObject(client client.Client, namespace string) error {
 
 	return client.Create(context.TODO(), backup)
 }
+
+func (r *ReconcileBackup) getAllHiveObjectsInNamespace(namespace string) ([]*hiveObject, error) {
+	nsLogger := addNamespaceLoggerFields(r.logger, namespace)
+
+	allNSObjs := []runtime.Object{}
+
+	allClusterDeployments := &hivev1.ClusterDeploymentList{}
```

Comment:
As with the watch code, make this generic by doing something like the following.
```
	for _, t := range typesToList {
		listObj := t.DeepCopyObject()
		if err := r.List(context.TODO(), listObj, client.InNamespace(namespace)); err != nil {
			nsLogger.WithError(err).Errorf("Couldn't get list of %s", t.GetObjectKind().GroupVersionKind().Kind)
			return nil, err
		}
		list, err := meta.ExtractList(listObj)
		if err != nil {
			// log something
			return nil, err
		}
		allNSObjs = append(allNSObjs, list...)
	}
```

Put a `typesToList` at the top of the file.
```
var (
	typesToList = []runtime.Object{
		&hivev1.ClusterDeploymentList{},
		&hivev1.SyncSetList{},
	}
)
```

Reply:
The include is 	"k8s.io/apimachinery/pkg/api/meta".